dfastmi/batch.py

```diff
@@ -139,56 +140,56 @@ def batch_mode_core(
             )
         )
     
-    branch = config["General"]["Branch"]
+    branch_name = config["General"]["Branch"]
     try:
-        ibranch = rivers.branches.index(branch)
+        ibranch = next((i for i, branch in enumerate(rivers.branches) if branch.name == branch_name), -1)
     except:
-        ApplicationSettingsHelper.log_text("invalid_branch", dict={"branch": branch}, file=report)
+        ApplicationSettingsHelper.log_text("invalid_branch", dict={"branch": branch_name}, file=report)
         success = False
     else:
-        reach = config["General"]["Reach"]
+        reach_name = config["General"]["Reach"]
         try:
-            ireach = rivers.allreaches[ibranch].index(reach)
+            ireach = next((i for i, reach in enumerate(rivers.branches[ibranch].reaches) if reach.name == reach_name), -1)
+            reach = rivers.branches[ibranch].reaches[ireach]
```

Comment:
why the Ireach, cant you directly get the reach?

Reply:
I agree, but batch.py is not in scope of this issue. I only refactor code to get it working again. Will pickup the good comments later in issue DFAST-103 or DFAST-104

Reply:
Discussed, we will leave these comments for a later refactoring.
When a new issue is created for this, the number will be disclosed here.